src/qamqpclient_p.h

```diff
@@ -100,6 +101,12 @@ class QAMQP_EXPORT QAmqpClientPrivate : public QAmqpMethodFrameHandler
     QAMQP::Error error;
     QString errorString;
 
+    /*! Exchange objects */
+    QAmqpChannelHash exchanges;
```

Comment:
What happens if a user declares an exchange named `foo`, and binds a queue named `foo`?  I'm not sure if that's against the AMQP specs but if permitted, then we'd have two different objects with the same name.

This handles that possibility in the cleanest way I can think of.  I figured it was easier to do this than to try and find out if the other was forbidden. :-)